core/src/test/java/org/testcontainers/containers/output/NewLineContainerTest.java

```diff
@@ -0,0 +1,38 @@
+package org.testcontainers.containers.output;
+
+import static org.junit.Assert.assertFalse;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.Test;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.GenericContainer;
+
+// Based on an idea by @ReuDa in
+// https://github.com/testcontainers/testcontainers-java/issues/1854#issuecomment-703459478.
+public class NewLineContainerTest {
+
+    private static final AlpineContainer container = new AlpineContainer();
+
+    @Test
+    public void newlines_are_not_added_to_container_output() throws Exception {
+        container.start();
+
+        String largeString = RandomStringUtils.randomAlphabetic(10000);
+        assertFalse(largeString.contains("\n"));
+
+        ExecResult build = container.execInContainer("echo", "-n", largeString);
+        assertFalse(build.getStdout().contains("\n"));
+    }
```

Comment:
I tried to add a `getLogs`-based test here as well, but it was surprisingly hard since I needed to override the `withCommand` etc to generate some log output to get that working. I tried something like this to no avail.

I'm willing to live with this, but it's up to the project maintainers to make the final call.

```java
package org.testcontainers.containers.output;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.Assert.assertFalse;

import org.apache.commons.lang.RandomStringUtils;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.testcontainers.containers.Container.ExecResult;
import org.testcontainers.containers.GenericContainer;

// Based on an idea by @ReuDa in
// https://github.com/testcontainers/testcontainers-java/issues/1854#issuecomment-703459478.
public class NewLineContainerTest {

    private static final AlpineContainer container = new AlpineContainer();

    @Before
    public void setUp() {
        container.start();
    }

    @After
    public void tearDown() {
        container.stop();
    }

    @Test
    public void newlines_are_not_added_to_getLogs_output() throws Exception {
        assertThat(container.getLogs()).doesNotContain("\n");
    }

    @Test
    public void newlines_are_not_added_to_execInContainer_output() throws Exception {
        String largeString = RandomStringUtils.randomAlphabetic(10000);
        assertFalse(largeString.contains("\n"));

        ExecResult result = container.execInContainer("echo", "-n", largeString);

        assertThat(result.getStdout()).doesNotContain("\n");
    }

    static class AlpineContainer extends GenericContainer<AlpineContainer> {
        AlpineContainer() {
            super("alpine");
        }

        @Override
        protected void configure() {
            super.configure();

            String largeString = RandomStringUtils.randomAlphabetic(10000);
            assertFalse(largeString.contains("\n"));

            this.withCommand("sh", "-c", "'echo -n " + largeString + "; sleep 2m'");
        }
    }
}
```